### PR TITLE
Added deprecation warning for `.label_encoding()`

### PR DESCRIPTION
**Pull request by mayankanand007**
This PR addresses issue #8608 by adding a deprecation warning before we remove the functionality entirely.

### CODE REVIEW

File: python/cudf/cudf/core/series.py [hunk 1]
```diff
@@ -3636,6 +3636,12 @@ def label_encoding(self, cats, dtype=None, na_sentinel=-1):
         4   -1
         dtype: int8
         """
+        
+        warnings.warn(
+            "Series.label_encoding is deprecated and will be removed "
+            "in the future.",
```
Comment (shwina) [inline python/cudf/cudf/core/series.py:3642]: Is there an alternative approach we want to suggest here?

Comment (mayankanand007) [reply python/cudf/cudf/core/series.py:3642]: According to issue description, the core functionality is already present in the form of `LabelEncoder` in `cuML`. Is there something else that is planned? I assumed the end goal is to remove the functionality altogether.

Comment (shwina) [reply python/cudf/cudf/core/series.py:3642]: That's perfect! Would suggest we add that in the warning message: e.g., `"Series.label_encoding is deprecated and will be removed in the future. Consider using cuML's LabelEncoder instead."`.

Comment (mayankanand007) [reply python/cudf/cudf/core/series.py]: Yeah that sounds better, thanks (and updated)!

**Review by shwina — APPROVED** [review]

**Comment by shwina** [comment]
@gpucibot merge

**Pull request merged**